packages/react/src/transactions/adapters/responders/UpdateProfileImageResponder.ts

```diff
@@ -0,0 +1,67 @@
+import { ApolloClient, NormalizedCacheObject } from '@apollo/client';
+import {
+  GetProfileDocument,
+  GetProfileQuery,
+  GetProfileQueryVariables,
+  MediaSetFragment,
+} from '@lens-protocol/api-bindings';
+import {
+  UpdateOffChainProfileImageRequest,
+  UpdateProfileImageRequest,
+} from '@lens-protocol/domain/use-cases/profile';
+import {
+  BroadcastedTransactionData,
+  ITransactionResponder,
+} from '@lens-protocol/domain/use-cases/transactions';
+
+function isUpdateOffChainProfileImageRequest(
+  request: UpdateProfileImageRequest,
+): request is UpdateOffChainProfileImageRequest {
+  return 'url' in request;
+}
+
+export class UpdateProfileImageResponder
+  implements ITransactionResponder<UpdateProfileImageRequest>
+{
+  constructor(private readonly client: ApolloClient<NormalizedCacheObject>) {}
+
+  async prepare({ request }: BroadcastedTransactionData<UpdateProfileImageRequest>) {
+    if (isUpdateOffChainProfileImageRequest(request)) {
+      const profileIdentifier = this.client.cache.identify({
+        __typename: 'Profile',
+        id: request.profileId,
+      });
+      this.client.cache.modify({
+        id: profileIdentifier,
+        fields: {
+          picture(oldPicture: MediaSetFragment) {
+            return {
+              ...oldPicture,
+              original: {
+                ...oldPicture.original,
+                url: request.url,
+              },
+            };
+          },
+        },
+      });
+    }
+  }
+
+  async commit(_: BroadcastedTransactionData<UpdateProfileImageRequest>) {
+    //
```

Comment:
In other places we do perform a profile query with `network-only` fetch policy so to update the cache. API might return different URLs to optimised media. The bigger argument is for publication media so cannot predict if and when this will be extended to profile image.